bluesky/callbacks/core.py

```diff
@@ -686,13 +686,11 @@ def descriptor(self, doc):
         """Write the header for the actual scan data"""
         # List all scalar fields, excluding the motor (x variable).
         self._read_fields = sorted([k for k, v in doc['data_keys'].items()
-                                    if (v['object_name'] != self._motor
```

Comment:
There is no guarantee that *any* data key will match `self._motor`. In some circumstances, `motor.user_readback.name == 'motor'`, but in other cases it might not. The only general way to handle this to come object names with object names, not object names with data keys.